tee-worker/litentry/core/assertion-build/src/nodereal/spaceid/mod.rs

```diff
@@ -0,0 +1,226 @@
+// Copyright 2020-2023 Trust Computing GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+extern crate sgx_tstd as std;
+
+pub mod bnb_domain_holding_amount;
+
+use crate::*;
+use lc_data_providers::{
+	nodereal::{DomainInfo, NoderealClient, SpaceIDApiList},
+	DataProviderConfigReader, ReadDataProviderConfig,
+};
+use serde::{Deserialize, Serialize};
+use std::string::ToString;
+
+pub struct SPACEIDLogic;
+pub trait SPACEIDLogicInterface {
+	fn get_bnb_domain_holding_amount(addresses: &[String]) -> Result<usize>;
+}
+
+impl SPACEIDLogicInterface for SPACEIDLogic {
+	fn get_bnb_domain_holding_amount(owners: &[String]) -> Result<usize> {
+		let config = DataProviderConfigReader::read()
+			.map_err(|e| Error::RequestVCFailed(Assertion::BnbDomainHolding, e))?;
+		let mut client = NoderealClient::new(&config);
+		let response = client.by_owners(owners).map_err(|e| {
+			Error::RequestVCFailed(Assertion::BnbDomainHolding, e.into_error_detail())
+		})?;
+
+		let owned_domains: Domains = Domains::from_value(&response)
+			.map_err(|_e| Error::UnclassifiedError(ErrorDetail::ParseError))?;
```

Comment:
We should avoid using `UnclassifiedError` when possible, as commented in the enum:

> 	// should be unreached, but just to be on the safe side
	// we should classify the error if we ever get this